code/_onclick/hud/parallax.dm

```diff
@@ -0,0 +1,255 @@
+/*
+ * This file handles all parallax-related business once the parallax itself is initialized with the rest of the HUD
+ */
+
+var/list/parallax_on_clients = list()
+
+/obj/screen/parallax
+	var/base_offset_x = 0
+	var/base_offset_y = 0
+	mouse_opacity = 1
+	icon = 'icons/turf/space.dmi'
+	icon_state = "blank"
+	name = "space parallax"
+	blend_mode = BLEND_ADD
+	layer = AREA_LAYER
+	plane = PLANE_SPACE_PARALLAX//changing this var doesn't actually change the plane of its overlays
+	var/parallax_speed = 0
+
+/obj/screen/plane_master
+	appearance_flags = PLANE_MASTER
+	screen_loc = "1,1"
+
+/obj/screen/plane_master/parallax_master
+	plane = PLANE_SPACE_PARALLAX
+	color = list(
+	1, 0, 0, 0,
+	0, 1, 0, 0,
+	0, 0, 1, 0,
+	0, 0, 0, 0,
+	0, 0, 0, 1)
+
+/obj/screen/plane_master/parallax_dustmaster
+	plane = PLANE_SPACE_DUST
+
+/datum/hud/proc/update_parallax()
+	var/client/C = mymob.client
+	if(!parallax_initialized || C.updating_parallax) return
+
+	for(var/turf/T in range(get_turf(C.eye),C.view))
```

Comment:
This should be `for(var/turf/space/T in range(get_turf(C.eye), C.view)) - at the moment you're doing an istype() on everything in the range twice... once to see if it's a `/turf` and then again to see if it's a `turf/space`